pkg/agent/rest.go

```diff
@@ -28,33 +30,40 @@ type NodeZeroRestClient struct {
 
 // NewNodeZeroRestClient Initialize a new rest client to interact with the Agent Rest API on node zero.
 func NewNodeZeroRestClient(ctx context.Context, assetDir string) (*NodeZeroRestClient, error) {
+
 	restClient := &NodeZeroRestClient{}
+	agentManifests := &manifests.AgentManifests{}
+	agentConfigAsset := &agentconfig.AgentConfig{}
 
 	assetStore, err := assetstore.NewStore(assetDir)
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to create asset store")
 	}
-	nmState := &manifests.NMStateConfig{}
-	if err := assetStore.Fetch(nmState); err != nil {
-		return nil, errors.Wrapf(err, "failed to fetch %s", nmState.Name())
+
+	if agentConfigError := assetStore.Fetch(agentConfigAsset); agentConfigError != nil {
```

Comment:
It looks to me like `Load()` might be closer to what we want here:
https://github.com/lranjbar/installer/blob/63c6e934ee6e33a4f2a5c63665a959aab58ca2a0/pkg/asset/store.go#L18-L20